homeassistant/components/hyperion/__init__.py

```diff
@@ -210,6 +321,13 @@ async def async_unload_entry(
         config_data = hass.data[DOMAIN].pop(config_entry.entry_id)
         for func in config_data[CONF_ON_UNLOAD]:
             func()
+
+        # Disconnect the shared instance clients.
+        for instance_num in list(config_data[CONF_INSTANCE_CLIENTS]):
+            instance_client = config_data[CONF_INSTANCE_CLIENTS][instance_num]
+            await instance_client.async_client_disconnect()
```

Comment:
I notice you have a couple of cases an `await` inside a for-loop. In such cases it is worth considering if this could be done in parallel with an `await asyncio.gather([X, Y]))`

Reply:
Thanks, good point. Fixed here. I found one other spot (`async_instances_to_clients_raw`) from a quick scan, but I think it would impede readability quite a bit and the n-count is likely 1-2, so I'm gauging it not worth it. Let me know if you disagree.